posthog/caching/tolerant_zlib_compressor.py

```diff
@@ -30,17 +31,22 @@ class TolerantZlibCompressor(BaseCompressor):
     """
 
     # we don't want to compress all values, e.g. feature flag cache in decide is already small
-    min_length = 1024
-    preset = 6
+    min_length = 512
+    zstd_preset = 0
+    zstd_threads = 1
+    zlib_preset = 6
 
     def compress(self, value: bytes) -> bytes:
         if settings.USE_REDIS_COMPRESSION and len(value) > self.min_length:
-            return zlib.compress(value, self.preset)
+            return zstd.compress(value, self.zstd_preset, self.zstd_threads)
         return value
 
     def decompress(self, value: bytes) -> bytes:
         try:
-            return zlib.decompress(value)
+            try:
+                return zstd.decompress(value)
+            except zstd.Error:
+                return zlib.decompress(value)  # Phasing out zlib, it is 10x slower and compresses worse
```

Comment:
we could add a prom counter here (fine as a follow-up) so we know when we're handling zlib and can decide when to remove it - but not blocking... better to get this out and check the impact